ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingIntegrationTest.kt

```diff
@@ -0,0 +1,70 @@
+package co.electriccoin.zcash.ui.screen.onboarding.view
+
+import androidx.compose.ui.test.junit4.StateRestorationTester
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.test.filters.MediumTest
+import co.electriccoin.zcash.ui.R
+import co.electriccoin.zcash.ui.screen.onboarding.TestOnboardingActivity
+import co.electriccoin.zcash.ui.screen.onboarding.model.OnboardingStage
+import co.electriccoin.zcash.ui.test.getStringResource
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+// TODO [#382]: https://github.com/zcash/secant-android-wallet/issues/382
+class OnboardingIntegrationTest {
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<TestOnboardingActivity>()
+
+    private fun newTestSetup(initialStage: OnboardingStage) = OnboardingTestSetup(composeTestRule, initialStage)
+
+    /**
+     * The test semantics are built upon StateRestorationTester component. We simulate screen state
+     * restoration with method emulateSavedInstanceStateRestore(), which needs to have setContent()
+     * method called beforehand. Then, after state restores after emulateSavedInstanceStateRestore(),
+     * setContent() callback is called again. Thus we can null the testSetup variable, call state
+     * restoration, and then make the assertion on it again.
+     */
+    @Test
+    @MediumTest
+    fun current_stage_restoration() {
+        val restorationTester = StateRestorationTester(composeTestRule)
+        var testSetup: OnboardingTestSetup? = null
+
+        restorationTester.setContent {
+            testSetup = newTestSetup(OnboardingStage.UnifiedAddresses)
```

Comment:
As-is, I think this line means that the assertions below can never fail because we re-set the state here.  I'm pushing a commit now that I think will resolve it.

Reply:
Oh yes, my bad. Thank you for the review. Now it's good to go.